### PR TITLE
auto update release branch ondemand e2e

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

This PR will add new workflow for update release branch when new branch is created similar we do with renovate, but
inside ondemand pipeline.

[K8S-9688](https://dt-rnd.atlassian.net/browse/K8S-9688)

> Sidenote:
> I have introduced a new yaml lib because Pyyaml doesn't support new standards and cannot parse properly GH action yaml files.

## How can this be tested?

1. Run git command:
`git branch -r --list 'origin/release-*' | sort --version-sort | tail -n 1| tr -d ' ' > release-branches.txt`
2. Update https://github.com/dynatrace/dynatrace-operator/blob/959ec573813a8a02f4b50e74c32dbcebb62fc456/.github/workflows/e2e-tests-ondemand.yaml#L12 to `release-1.0`
3. `python hack/build/ci/update-e2e-ondemand-pipeline.py release-branches.txt .github/workflows/e2e-tests-ondemand.yaml`
4. Make sure default branch in `.github/workflows/e2e-tests-ondemand.yaml`: 
```
...
  workflow_dispatch:
    inputs:
      target:
        description: Target branch to run E2E tests over
        required: true
        default: release-1.3
...
```

### CODE REVIEW

File: .github/workflows/self-update-on-new-release-version.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          add-paths: |
+            .github/renovate.json5
+          title: '[Automatic] Update .github/workflows/e2e-tests-ondemand.yaml config for ${{ github.ref }}'
+          labels: 'ci'
+          commit-message: Update e2e-tests-ondemand.yaml
+          committer: GitHub <noreply@github.com>
+          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          body: |
+            # Description
+            Upgrade `e2e-tests-ondemand.yaml` to include last release branche as default.
```
Comment (waodim) [inline .github/workflows/self-update-on-new-release-version.yaml:52]:
```suggestion
            Upgrade `e2e-tests-ondemand.yaml` to include latest release branch as default.
```

**Review by waodim — COMMENTED** [review]
Apart from minor nitpick lgtm and tested it - works for me

File: .github/workflows/self-update-on-new-release-version.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: Self update on new release
+on:
+  push:
+    branches:
+      - release-*
+
+jobs:
+  update-e2e-ondemand-release:
```
Comment (chrismuellner) [inline .github/workflows/self-update-on-new-release-version.yaml:8]: this is really similar to the Renovate workflow we have, so we should combine those and only have a single PR when a new release branch is created.

File: .github/workflows/self-update-on-new-release-version.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          commit-message: Update e2e-tests-ondemand.yaml
+          committer: GitHub <noreply@github.com>
+          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          body: |
+            # Description
+            Upgrade `e2e-tests-ondemand.yaml` to include last release branche as default.
+            ## How can this be tested?
+            Check renovate app.
+            ## Checklist
+            - [x] PR is labeled accordingly
```
Comment (chrismuellner) [inline .github/workflows/self-update-on-new-release-version.yaml:56]: why even have the checklist here then?

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: .github/workflows/self-update-on-new-release-version.yaml (hunk 2 above)
Comment (andriisoldatenko) [reply .github/workflows/self-update-on-new-release-version.yaml:8]: it's a good idea, if you want I can do this way, just was not sure do we want same workflow or different.

File: .github/workflows/self-update-on-new-release-version.yaml (hunk 3 above)
Comment (andriisoldatenko) [reply .github/workflows/self-update-on-new-release-version.yaml:56]: copy/pasta from sample 👍

File: .github/workflows/self-update-on-new-release-version.yaml (hunk 2 above)
Comment (chrismuellner) [reply .github/workflows/self-update-on-new-release-version.yaml:8]: we should only have a single workflow that updates files for the release, so let's combine it

Comment (andriisoldatenko) [reply .github/workflows/self-update-on-new-release-version.yaml:8]: do you have idea how to name it? 
`new-release`?

Comment (andriisoldatenko) [reply .github/workflows/self-update-on-new-release-version.yaml:8]: fixed in 4d1a38e0

File: .github/workflows/self-update-on-new-release-version.yaml (hunk 1 above)
Comment (andriisoldatenko) [reply .github/workflows/self-update-on-new-release-version.yaml]: fixed in 11e9d891

File: .github/workflows/self-update-on-new-release-version.yaml (hunk 3 above)
Comment (andriisoldatenko) [reply .github/workflows/self-update-on-new-release-version.yaml]: fixed in 11e9d891

**Review by waodim — APPROVED** [review]
lgtm

**Comment by andriisoldatenko** [comment]
@chrismuellner PTAL

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**